src/Orchard.Web/Modules/Orchard.Localization/Services/LocalizationService.cs

```diff
@@ -56,18 +56,24 @@ public class LocalizationService : ILocalizationService {
 
         IEnumerable<LocalizationPart> ILocalizationService.GetLocalizations(IContent content) {
             // Warning: May contain more than one localization of the same culture.
-            return ((ILocalizationService) this).GetLocalizations(content, null);
+            return ((ILocalizationService)this).GetLocalizations(content, null);
         }
 
         IEnumerable<LocalizationPart> ILocalizationService.GetLocalizations(IContent content, VersionOptions versionOptions) {
             if (content.ContentItem.Id == 0)
                 return Enumerable.Empty<LocalizationPart>();
-
             var localized = content.As<LocalizationPart>();
-
-            var query = versionOptions == null
-                ? _contentManager.Query<LocalizationPart>(localized.ContentItem.ContentType)
-                : _contentManager.Query<LocalizationPart>(versionOptions, localized.ContentItem.ContentType);
+            IContentQuery<LocalizationPart> query;
+            if (content.ContentItem.TypeDefinition.Parts.Any(x => x.PartDefinition.Name == "TermPart")) { // terms translations can be contained on different TermContentType linked to taxonomies translations
```

Comment:
@MatteoPiovanelli-Laser @sebastienros 

I just found this during merging 1.10.x to dev and I have to admit that I looked at this PR and didn't notice this, but this is really bad. We can't put a reference like this in a module that provides low-level functionality (it's not a dependency, so it doesn't break anything, but still).

What's really strange is that two years later, @HermesSbicego-Laser committed a change that removes this, then two commits later I merged 1.10.x into dev, which readded that change. I don't know how the difftool would think that that should happen, but re-adding removed code by accident (without noticing) is even worse. This is not the first time it happens (and merging from 1.10.x to dev caused a lot of different problems in other cases too) - suggestions are welcome. I'll try to sort out the correct logic for LocalizationService...

Reply:
Update: I made a mistake, Hermes didn't remove this (I was looking at the wrong diff). Nevertheless, this TermPart-related logic has to be removed somehow without breaking Taxonomy Localization.

Reply:
Removed the TermPart condition in this commit: a5a24d07b70da97adf89f9d3e48ac27a40f6a807, so there's no type filter in the query. This way it doesn't Taxonomies localization and I don't think filtering on content type matters anyway.